crypto/src/ed25519.rs

```diff
@@ -64,6 +65,7 @@ impl<'a> From<&'a Ed25519PrivateKey> for Ed25519PublicKey {
 impl VerifyingKey for Ed25519PublicKey {
     type PrivKey = Ed25519PrivateKey;
     type Sig = Ed25519Signature;
+    type Bytes = Ed25519PublicKeyBytes;
```

Comment:
Why?

Reply:
I think one thing is that without defining this, we cannot map a PK <-> it's corresponding PK bytes form within the VerifyingKeyTrait. One thing I added in this PR is the trait bound `TryFrom<Self::Bytes>` on VerifyingKeyBytes - I have tried other ways but I personally don't understand how to define this without an associated type.

Reply:
I personally think this trait is very useful, but I could be wrong

Reply:
> without defining this, we cannot map a PK <-> it's corresponding PK bytes form within the VerifyingKeyTrait

OK, why do you want to do this?

> we can't define functions that takes in some generic PublicKey, and also interacts with the PublicKeyBytes

Do you have an example of such a function?